test/e2e/storage/sanity.go

```diff
@@ -504,25 +521,30 @@ func unpublishVolume(s csi.ControllerClient, c csi.NodeClient, sc *sanity.Sanity
 	Expect(nodeunstagevol).NotTo(BeNil())
 }
 
-func deleteVolume(s csi.ControllerClient, vol *csi.Volume) {
+func deleteVolume(s csi.ControllerClient, vol *csi.Volume, volName string, cl *sanity.Cleanup) {
 	var err error
 
-	By("cleaning up deleting the volume")
+	By("Deleting the volume: " + vol.GetVolumeId())
 	_, err = s.DeleteVolume(
 		context.Background(),
 		&csi.DeleteVolumeRequest{
 			VolumeId: vol.GetVolumeId(),
 		},
 	)
+	if err != nil {
+		By("Deleting volume  " + vol.GetVolumeId() + " reply: " + err.Error())
+	}
+
+	cl.UnregisterVolume(volName)
 	framework.ExpectNoError(err, "delete volume %s", vol.GetVolumeId())
 }
 
-var unreachable = v1.Taint{Key: "node.kubernetes.io/unreachable", Effect: "NoSchedule"}
+//var unreachable = v1.Taint{Key: "node.kubernetes.io/unreachable", Effect: "NoSchedule"}
 
 // restartNode works only for one of the nodes in the QEMU virtual cluster.
 // It does a hard poweroff via SysRq and relies on Docker to restart the
 // "failed" node.
-func restartNode(cs clientset.Interface, nodeID string) {
+func restartNode(cs clientset.Interface, nodeID string, sc *sanity.SanityContext) {
```

Comment:
nit: if this is not needed, it can be removed entirely.

Reply:
If I understood correctly your comment is about above change (line #542), yes we can remove

Reply:
Yes.